src/odemis/driver/technolution.py

```diff
@@ -659,100 +672,151 @@ def _getScanStepSizeBits(self):
+            raise ValueError("Incorrect timing between the setpoints has been found for acquisition in x direction")
 
-        # Slope of the scan gain in bits per second
-        scan_gain_slope = (scan_gain_bits - scan_offset_bits) / scanning_time
-        scanning_points = scan_gain_slope * time_setpoints_scan + scan_offset_bits  # x_setpoints=slope*time + offset
+        # Slope of the scanning ramp in bits per second
+        scanning_slope = (scan_gain_bits - scan_offset_bits) / scanning_time
+        scanning_points = scanning_slope * timestamp_setpoints + scan_offset_bits  # x_setpoints=slope*time + offset
 
+        # Use almost_equal to handle floating point errors.
         if not almost_equal(remainder_scanning_time, 0, rtol=0, atol=1e-10):
             # Allow the descanner to finish scanning the last pixels by adding one setpoint with the scan gain value.
             scanning_points = numpy.hstack((scanning_points, scan_gain_bits))
-            # Extra setpoint can be added at the maximum entered value (scan_gain) which is approved by Wilco
-            # and Andries. Because of the inertia of the mirror it is expected that this is the best solution
-            # (relatively fast and safe)
+            # Note: For now the extra setpoint to compensate for the remaining scanning time has a value equal to the
+            # scan gain. Because of the inertia of the mirror it is expected that this is a decent solution. (
+            # relatively fast and safe)
 
         # Rounding needed to prevent increase of length due to floating point errors.
-        if len(scanning_points) != math.ceil(numpy.round((dwellTime * X_cell_resolution) / descan_period, 10)):
+        if len(scanning_points) != math.ceil(numpy.round((dwellTime * x_cell_resolution) / descan_period, 10)):
             raise ValueError("Incorrect number of setpoints defined for acquisition in x direction")
 
-        # Calculation flyback_points:
-        # Use math.ceil so that if the physicalFlybackTime is not a whole multiple of the descan period the flyback
-        # points allow enough time the descanner to move back.
-        flyback_points = scan_offset_bits + numpy.zeros(math.ceil(self.physicalFlybackTime / descan_period))
+        # Calculation of the flyback points:
+        # Round up so that if the physical flyback time (physical restricted time for the mirror to move back to its
+        # original position) is not a whole multiple of the descan period the flyback points allow enough time for the
+        # descanner to move back.
+        number_flyback_points = math.ceil(self.physicalFlybackTime / descan_period)
+        flyback_points = scan_offset_bits + numpy.zeros(number_flyback_points)
 
         setpoints = numpy.concatenate((scanning_points, flyback_points))
 
         if setpoints.min() < INT16.min or setpoints.max() > INT16.max:
             raise ValueError("Acquisition setpoints for the descanner are out of range as excepted by the ASM API.")
```

Comment:
Same here. Would will we do if this error is raised. We don't have a backup as for VAs when you want to set a value out of range. I think the gain VA and offset VA already should make sure that the values are in range.

Reply:
Check if conversion to Delmic gain makes this check unneeded?

Reply:
Indeed this is not needed.